internal/ingress/controller/parser/parser.go

```diff
@@ -1351,7 +1351,9 @@ func (p *Parser) getCerts(secretsToSNIs map[string][]string) []Certificate {
 				CreationTimestamp: secret.CreationTimestamp,
 			}
 		} else {
-			if kongCert.CreationTimestamp.After(secret.CreationTimestamp.Time) {
+			if kongCert.CreationTimestamp.After(secret.CreationTimestamp.Time) ||
+				(kongCert.CreationTimestamp.Equal(&secret.CreationTimestamp) &&
+					*kongCert.cert.ID > string(secret.UID)) {
```

Comment:
Also, please add a test for this condition (under `func TestParserSecret`)

Reply:
ok.
A new PR has been committed to the next branch